src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -2977,15 +2977,15 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
                 std::string type_code = ASRUtils::get_type_code(asr_tuple->m_type,
                                                                 asr_tuple->n_type);
                 std::vector<llvm::Type*> llvm_el_types;
+                ASR::storage_typeType m_storage = ASR::storage_typeType::Default;
+                bool is_array_type = false, is_malloc_array_type = false;
+                bool is_list = false;
+                ASR::dimension_t* m_dims = nullptr;
+                int n_dims = 0, a_kind = -1;
                 for( size_t i = 0; i < asr_tuple->n_type; i++ ) {
-                    bool is_local_array_type = false;
-                    int local_n_dims = 0;
-                    int local_a_kind = -1;
-                    ASR::storage_typeType local_m_storage = ASR::storage_typeType::Default;
-                    llvm_el_types.push_back(get_arg_type_from_ttype_t(asr_tuple->m_type[i], m_abi,
-                                            arg_m_abi, local_m_storage, arg_m_value_attr, local_n_dims,
-                                            local_a_kind, is_local_array_type,
-                                            ASRUtils::intent_local)->getPointerElementType());
+                    llvm_el_types.push_back(get_type_from_ttype_t(asr_tuple->m_type[i],
```

Comment:
You can use `get_type_from_ttype_t_util` as well. It won't require you to declare auxiliary argument variables.

Reply:
Thanks! Updated